src/V3Delayed.cpp

```diff
@@ -76,140 +114,174 @@
 VL_DEFINE_DEBUG_FUNCTIONS;
 
 //######################################################################
-// Delayed state, as a visitor of each AstNode
+// Convert AstAssignDlys (NBAs)
 
 class DelayedVisitor final : public VNVisitor {
     // TYPES
 
-    // Components of an AstAssignDly LHS expression, e.g.:
-    //   Sel(ArraySel(ArraySel(_: VarRef, _: Index), _: Index), _: Lsb, _: Width))
-    struct LhsComponents final {
-        AstVarRef* refp = nullptr;  // The referenced variable
-        std::vector<AstNodeExpr*> arrIdxps;  // The array indices applied to 'refp'
-        AstNodeExpr* selLsbp = nullptr;  // The bit select LSB expression, after the array selects
-        AstConst* selWidthp = nullptr;  // The width of the bit select
+    // The various NBA conversion schemes, including error cases
+    enum class Scheme : uint8_t {
+        Undecided = 0,
+        UnsupportedCompoundArrayInLoop,
+        ShadowVar,
+        FlagShared,
+        FlagUnique,
+        ValueQueueWhole,
+        ValueQueuePartial
+    };
+
+    // All info associated with a variable that is the target of an NBA
+    class VarScopeInfo final {
+    public:
+        // First write reference encountered to the VarScope as the target on an NBA
+        const AstVarRef* firstNbaRefp = nullptr;
+        // Active block 'firstNbaRefp' is under
+        const AstActive* fistActivep = nullptr;
+        bool partial = false;  // Used on LHS of NBA under a Sel
+        bool inLoop = false;  // Used on LHS of NBA in a loop
+        bool inSuspOrFork = false;  // Used on LHS of NBA in suspendable process or fork
+        Scheme scheme = Scheme::Undecided;  // Conversion scheme to use for this variable
+        uint32_t nTmp = 0;  // Temporary number for unique names
+
+    private:
+        // Combined sensitivities of all NBAs targeting this variable
+        AstSenTree* m_senTreep = nullptr;
+
+        // Union of stuff needed for the various schemes - use accessors below!
+        union {
+            struct {  // Stuff needed for Scheme::ShadowVar
+                AstVarScope* vscp;  // The shadow variable
```

Comment:
Usually we use `m_` naming even if public variable (can they not be public?), but I'm not super-stickler on this if only used inside a containing class.

The below unions should have `m_` added though as seem all private.

Reply:
added `m_`, but trivial getters/setters on a data structure local to the file seem a bit tedious.

Reply:
Thanks.  No need for getter/setters, I also avoid them on little containers too.